src/test/url-loader/package-url-resolver_test.ts

```diff
@@ -70,9 +70,37 @@ suite('PackageUrlResolver', function() {
       assert.equal(
           r.resolve(packageRelativeUrl`http://abc.xyz/foo.html`),
           resolvedUrl`http://abc.xyz/foo.html`);
+
+      assert.equal(
+          r.resolve(
+              resolvedUrl`https://baz.com/qux.js`,
+              fileRelativeUrl`https://foo.com/bar.html`),
+          resolvedUrl`https://foo.com/bar.html`);
+    });
+
+    test('resolves protocol-relative URLs using default protocol', () => {
+      const r = new PackageUrlResolver();
       assert.equal(
           r.resolve(packageRelativeUrl`//abc.xyz/foo.html`),
-          resolvedUrl`file://abc.xyz/foo.html`);
+          resolvedUrl`https://abc.xyz/foo.html`);
+
+      assert.equal(
+          r.resolve(
+              resolvedUrl`http://foo.com/bar.html`,
+              fileRelativeUrl`//foo.com/bar.html`),
+          resolvedUrl`https://foo.com/bar.html`);
+    });
+    test('resolves protocol-relative URLs using provided protocol', () => {
+      const r = new PackageUrlResolver({protocol: 'potato:'});
+      assert.equal(
+          r.resolve(packageRelativeUrl`//abc.xyz/foo.html`),
+          resolvedUrl`potato://abc.xyz/foo.html`);
+
+      assert.equal(
+          r.resolve(
+              resolvedUrl`https://foo.com/bar.html`,
+              fileRelativeUrl`//foo.com/bar.html`),
+          resolvedUrl`potato://foo.com/bar.html`);
```

Comment:
We should also test how protocol and host interact. I think this is the correct behavior:

```typescript
       const resolver = new PackageUrlResolver(
           {componentDir: `components`, host: `abc.xyz`, packageDir: `/1/2`, protocol: 'carrot'});
       assert.equal(
           resolver.resolve(packageRelativeUrl`//abc.xyz/foo.html`),
           rootedFileUrl`1/2/foo.html`);
```